OSDOCS#14791:Update the z-stream RN for 4.18.16 (#94134)
openshift/openshift-docs: release_notes/ocp-4-18-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

> / 4.18.16

Reviewer: ```suggestion
// 4.18.16
```
showing in the output because only one slash

==== Bug fixes

* Previously, in non-Zonal Azure regions, a bug in the dynamically computed fault domain count update caused scaling failures. This issue prevented upgrades to {product-title} version 4.15.48 and later from scaling existing machine sets. With this release, an update is implemented that rectifies the dynamic fault domain calculation and prevents scaling failures in non-Zonal regions. This fix ensures smooth operation of machine set scaling after upgrading to {product-title} version 4.15.48 and later. (link:https://issues.redhat.com/browse/OCPBUGS-56654[OCPBUGS-56654])

* Previously, API server subject alternative name (SAN) validation was performed regardless of public key infrastructure (PKI) reconciliation status, causing potential connectivity issues with hosted control plane clusters due to invalid SANs. With this release, a fix removes validation for {product-title} API server SANs when PKI reconciliation is disabled. The hosted control plane performance is improved by eliminating unnecessary validation, especially when the PKI reconciliation is not managed. (link:https://issues.redhat.com/browse/OCPBUGS-56627[OCPBUGS-56627])

> * Previously, a bug in Bare Metal Operator (BMO) led to JSON parsing errors because of a missing Redfish system ID in Baseboard Management Controller (BMC) URLs. This issue caused users to receive errors when the system ID was left out of the URLs. With this release, BMO handles URLs without a Redfish system ID as addresses without a system ID. This fix improves the software's handling of missing a Redfish system ID in BMC URLs. (link:https://issues.redhat.com/browse/OCPBUGS-56431[OCPBUGS-56431])

Reviewer: ```suggestion
* Previously, a bug in the Bare Metal Operator (BMO) caused JSON parsing errors because of a missing Redfish system ID in Baseboard Management Controller (BMC) URLs. This issue caused users to receive errors when the system ID was left out of the URLs. With this release, BMO handles URLs without a Redfish system ID as addresses without a system ID. This fix improves software handling of missing a Redfish system ID in BMC URLs. (link:https://issues.redhat.com/browse/OCPBUGS-56431[OCPBUGS-56431])
```

> * Previously, during Ironic Python Agent (IPA) deployments, the absence of NetworkManager logs in RAM disk logs hinders effective debugging, impacting network issue resolution. With this release, NetworkManager logs are included in RAM disk logs for IPA debugging. This results in enhanced IPA logs that provide comprehensive NetworkManager data for improved debugging. (link:https://issues.redhat.com/browse/OCPBUGS-56097[OCPBUGS-56097])

Reviewer: ```suggestion
* Previously, during Ironic Python Agent (IPA) deployments, the absence of NetworkManager logs in RAM disk logs hinders effective debugging, impacting network issue resolution. With this release, NetworkManager logs are included in RAM disk logs for IPA debugging. This results in enhanced IPA logs that provide comprehensive `NetworkManager` data for improved debugging. (link:https://issues.redhat.com/browse/OCPBUGS-56097[OCPBUGS-56097])
```

> * Previously, during Ironic Python Agent (IPA) deployments, the absence of NetworkManager logs in RAM disk logs hinders effective debugging, impacting network issue resolution. With this release, NetworkManager logs are included in RAM disk logs for IPA debugging. This results in enhanced IPA logs that provide comprehensive `NetworkManager` data for improved debugging. (link:https://issues.redhat.com/browse/OCPBUGS-56097[OCPBUGS-56097])

Reviewer 2: I saw @ShaunaDiaz comment to go with `NetworkManager` data. However, I see very few instances of `NetworkManager` in the docs. If we keep the back ticks, should we go with `NetworkManager` logs?

> * Previously, the code did not support Docker image mirroring with a tag and digest. This resulted in failed Helm repository mirroring that caused image duplication and inconsistencies in deployment. With this release, a fix addresses Docker references in Helm repository mirroring that allows tag and digest,  and improves successful image mirroring. (link:https://issues.redhat.com/browse/OCPBUGS-56043[OCPBUGS-56043])

Reviewer: ```suggestion
* Previously, Helm did not support Docker image mirroring with a tag and digest. This resulted in failed Helm repository mirroring that caused image duplication and inconsistencies in deployment. With this release, a fix addresses Docker references in Helm repository mirroring that allows tag and digest, and improves successful image mirroring. (link:https://issues.redhat.com/browse/OCPBUGS-56043[OCPBUGS-56043])
```

* Previously, an issue started from an incorrect bucket name that was used for {op-system-first}. Users could not create {product-title} clusters because of a failed {op-system} image import. This problem was resolved by correcting the cluster creation in the Madrid zone for the `PowerVS` installer-provisioned infrastructure {cluster-capi-operator} by using the correct bucket name. With this release, the user can create {product-title} clusters in  the Madrid zone by using the installation program. (link:https://issues.redhat.com/browse/OCPBUGS-53142[OCPBUGS-53142])

> * Previously, intermittent resource leaks in the `MachineOSConfig` (MOSC) to `MachineOSBuild` (MOSB) connection due to missing owner references during job creation result in potential resource exhaustion, affecting pod updates. With this release, the owner references are added in the job creation to ensure the consistent removal of MOSB resources when MOSC is deleted, preventing resource leaks. (link:https://issues.redhat.com/browse/OCPBUGS-52189[OCPBUGS-52189])

Reviewer 2: ```suggestion
* Previously, intermittent resource leaks in the `MachineOSConfig` (MOSC) to `MachineOSBuild` (MOSB) connection due to missing owner references during job creation resulted in potential resource exhaustion, affecting pod updates. With this release, the owner references are added in the job creation to ensure the consistent removal of MOSB resources when MOSC is deleted, preventing resource leaks. (link:https://issues.redhat.com/browse/OCPBUGS-52189[OCPBUGS-52189])
```
